core/src/tangram.cpp

```diff
@@ -432,8 +427,16 @@ void Map::render() {
     // Run render-thread tasks
     impl->renderState.jobQueue.runJobs();
 
+    bool hasGeometryFeatureQuery = drawSelectionBuffer;
+    bool hasMarkerQuery = drawSelectionBuffer;
+
+    for (const auto& selectionQuery : impl->selectionQueries) {
+        if (selectionQuery.type() == QueryType::feature) { hasGeometryFeatureQuery = true; }
```

Comment:
The problem with `QueryType::label` is that they are draw in `onBeginDrawFrame`, that is commonly needed by all the styles.